lib/js/src/manager/SdlManager.js

```diff
@@ -0,0 +1,414 @@
+                }
+
+                if (this.getManagerListener() === null) {
+                    throw new Error('An SdlManagerListener must be defined');
+                }
+
+                this._lifecycleManager = new LifecycleManager(this.getAppConfig(), this._lifecycleListener);
+
+                /* FIXME: setSdlSecurity and related methods/classes do not exist
+                if (this._sdlSecList.length > 0) {
+                    this._lifecycleManager.setSdlSecurity(this._sdlSecList, this._serviceEncryptionListener);
+                }
+                */
+
+                this.initNotificationQueue();
+                this._lifecycleManager.start();
+            } else {
+                throw new Error('An AppConfig must be provided');
+            }
+        }
+        return this;
+    }
+
+    /**
+    * Gracefully disposes the managers and alerts and destroys the ManagerListener
+    * @param {RpcMessage} message
+    * @return {Promise} - A Promise which resolves if the RPC response is SUCCESS, otherwise rejects
+    */
+    dispose () {
+        if (this._permissionManager !== null) {
+            this._permissionManager.dispose();
+        }
+
+        if (this._fileManager !== null) {
+            this._fileManager.dispose();
+        }
+
+        if (this._screenManager !== null) {
+            this._screenManager.dispose();
+        }
+
+        if (this._lifecycleManager !== null) {
+            this._lifecycleManager.stop();
+        }
+
+        if (this._managerListener !== null) {
+            this._managerListener.onDestroy();
+            this._managerListener = null;
+        }
+
+        return this;
+    }
+
+    /**
+    * Sends a single RPC
+    * @param {RpcMessage} message
+    * @return {Promise} - A Promise which resolves if the RPC response is SUCCESS, otherwise rejects
+    */
+    async sendRpc (message) {
+        return this._lifecycleManager.sendRpcMessage(message);
+    }
+
+    /**
+    * Sends multiple RPCs asynchronously
+    * @param {RpcMessage[]} messages - An array of RpcMessages
+    * @return {Promise} - A Promise which resolves if all RPCs respond with SUCCESS, otherwise rejects with the first failure
+    */
+    async sendRpcs (messages) {
+        if (!Array.isArray(messages)) {
+            return Promise.reject('messages is not an array');
+        }
+        // Build an array of RPC Promises from the messages
+        const rpcPromiseArray = messages.map((message) => {
+            return this.sendRpc(message);
+        });
+        return Promise.all(rpcPromiseArray);
+    }
+
+    /**
+    * Sends multiple RPCs synchronously (in order)
+    * @param {RpcMessage[]} messages
+    * @return {Promise} - A Promise which resolves with the last RPC response if all RPCs respond with SUCCESS, otherwise rejects with the first failure
+    */
+    async sendSequentialRpcs (messages) {
+        return messages.reduce((chain, message) => {
```

Comment:
I don't believe this would execute one at a time. Has this been checked?

Reply:
Yep, it works! Someone wrote about it here, too: https://css-tricks.com/why-using-reduce-to-sequentially-resolve-promises-works/